packages/shorebird_cli/test/src/archive_analysis/ios_archive_differ_test.dart

```diff
@@ -21,148 +22,167 @@ void main() {
+                'ios-arm64_x86_64-simulator/App.framework/flutter_assets/assets/asset.json',
+                'ios-arm64/App.framework/_CodeSignature/CodeResources',
+                'ios-arm64/App.framework/flutter_assets/assets/asset.json'
+              },
+            );
+          });
         });
 
-        test('finds differences between two differed zipped xcframeworks', () {
-          expect(
-            differ
-                .changedFiles(baseXcframeworkPath, changedAssetXcframeworkPath)
-                .changedPaths,
-            {
-              'ios-arm64_x86_64-simulator/App.framework/_CodeSignature/CodeResources',
-              'ios-arm64_x86_64-simulator/App.framework/App',
-              'ios-arm64_x86_64-simulator/App.framework/flutter_assets/assets/asset.json',
-              'ios-arm64/App.framework/_CodeSignature/CodeResources',
-              'ios-arm64/App.framework/App',
-              'ios-arm64/App.framework/flutter_assets/assets/asset.json'
-            },
-          );
+        group('changedFiles', () {
+          test('detects asset changes', () {
+            final fileSetDiff = differ.changedFiles(
+              baseXcframeworkPath,
+              changedAssetXcframeworkPath,
+            );
+            expect(differ.assetsFileSetDiff(fileSetDiff), isNotEmpty);
+            expect(differ.dartFileSetDiff(fileSetDiff), isEmpty);
+            expect(differ.nativeFileSetDiff(fileSetDiff), isEmpty);
+          });
+
+          test('detects dart changes', () {
+            final fileSetDiff = differ.changedFiles(
+              baseXcframeworkPath,
+              changedDartXcframeworkPath,
+            );
+            expect(differ.assetsFileSetDiff(fileSetDiff), isEmpty);
+            expect(differ.dartFileSetDiff(fileSetDiff), isNotEmpty);
+            expect(differ.nativeFileSetDiff(fileSetDiff), isEmpty);
+          });
         });
       });
-
-      group('changedFiles', () {
-        test('detects asset changes', () {
-          final fileSetDiff = differ.changedFiles(
-            baseXcframeworkPath,
-            changedAssetXcframeworkPath,
-          );
-          expect(differ.assetsFileSetDiff(fileSetDiff), isNotEmpty);
-          expect(differ.dartFileSetDiff(fileSetDiff), isNotEmpty);
-          expect(differ.nativeFileSetDiff(fileSetDiff), isEmpty);
-        });
-
-        test('detects dart changes', () {
-          final fileSetDiff = differ.changedFiles(
-            baseXcframeworkPath,
-            changedDartXcframeworkPath,
-          );
-          expect(differ.assetsFileSetDiff(fileSetDiff), isEmpty);
-          expect(differ.dartFileSetDiff(fileSetDiff), isNotEmpty);
-          expect(differ.nativeFileSetDiff(fileSetDiff), isEmpty);
-        });
-      });
-    });
-  });
+    },
+    // These tests rely on the presence of the `codesign` executable, which is
+    // only present on macOS.
+    skip: !platform.isMacOS,
```

Comment:
unless we adjust our CI, this will never run right?

Reply:
tl;dr: not sure what to do here. Seems like overkill to spin up a new CI job specifically for a single test file.

Reply:
I'm not against setting up a matrix to run ci checks on all platforms fwiw

Reply:
I don't think this means we shouldn't set up the matrix, but I don't think that solves our problem here :( https://github.com/shorebirdtech/shorebird/actions/runs/5802441910/job/15728852901?pr=1065